src/pages/Transaction.vue

```diff
@@ -184,12 +171,13 @@ export default {
       }
     },
 
-    setTransaction(transaction) {
-      this.transaction = transaction
+    async updatePrice(transaction) {
+      this.price = await CryptoCompareService.dailyAverage(transaction.timestamp.unix)
     },
 
-    setAverage(average) {
-      this.average = average
+    setTransaction(transaction) {
+      this.transaction = transaction
+      this.initialBlockHeight = this.height - (this.transaction.confirmations + 1)
```

Comment:
```suggestion
      this.initialBlockHeight = this.height - this.transaction.confirmations
```

The `+ 1` isn't needed this way since the block height is set right away.